tests/UI/utils/setup.ts

```diff
@@ -47,25 +48,37 @@ after(async function () {
   }
 });
 
+const takeScreenShotAfterStep = async (browser: any, screenshotPath: string) => {
+  const currentTab = await utilsPlaywright.getLastOpenedTab(browser);
+
+  // Take a screenshot
+  if (currentTab !== null) {
+    await currentTab.screenshot(
+      {
+        path: screenshotPath,
+        fullPage: true,
+      },
+    );
+  }
+};
+
 /**
  * @function afterEach
  * @description Take a screenshot if a step is failed
  */
 afterEach(async function () {
   // Take screenshot if demanded after failed step
   if (global.SCREENSHOT.AFTER_FAIL && this.currentTest?.state === 'failed') {
-    const currentTab = await utilsPlaywright.getLastOpenedTab(this.browser);
-
-    // Take a screenshot
-    if (currentTab !== null) {
-      await currentTab.screenshot(
-        {
-          path: `${global.SCREENSHOT.FOLDER}/fail_test_${screenshotNumber}.png`,
-          fullPage: true,
-        },
-      );
-    }
-
+    await takeScreenShotAfterStep(this.browser, `${global.SCREENSHOT.FOLDER}/fail_test_${screenshotNumber}.png`);
     screenshotNumber += 1;
   }
+  if (global.SCREENSHOT.EACH_STEP) {
+    const testPath = this.currentTest?.file;
+    // eslint-disable-next-line no-unsafe-optional-chaining
+    const folderPath = testPath?.slice(testPath?.indexOf('tests/UI') + 8).slice(0, -3);
+    const screenshotPath = `${global.SCREENSHOT.FOLDER}${folderPath}/${_.kebabCase(this.currentTest?.title)}.png`;
```

Comment:
Add a function `kebabCase` to the `ui-testing-library` And update the library in the Core